package recipe for age.el

anticomputer opened the pull request:
### Brief summary of what the package does

age.el provides transparent age (https://github.com/FiloSottile/age) file encryption and decryption for Emacs in the style of EasyPG.

### Direct link to the package repository

https://github.com/anticomputer/age.el

### Your association with the package

Maintainer.

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

<!-- Please confirm by replacing `[]` with `[x]`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thanks for this.  Quick first pass --

`age.el` with byte-compile using Emacs 28.1:
```
age.el:69:29: Warning: assignment to free variable `age-inhibit'
In age-scrypt-p:
age.el:366:33: Warning: assignment to free variable `age-inhibit'
```
I think you can pacify the compiler in this case with `(defvar age-inhibit)`.

`age.el` with checkdoc 0.6.2:
```
age.el:469: Argument `input' should appear (as INPUT) in the doc string
```

I think the package boilerplate has some copy-paste errors, i.e.:
- age.el#L16: Have you done the paperwork to assign this copyright?
- age.el#L22: This may be a copy-paste error?


`age.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- age.el#L847: There is no need to quote lambdas (neither #' nor ')
- age.el#L882: It's safer to sharp-quote function names; use `#'`
- age.el#L1086: It's safer to sharp-quote function names; use `#'`
- age.el#L1271: It's safer to sharp-quote function names; use `#'`
- age.el#L1283: It's safer to sharp-quote function names; use `#'`
- age.el#L1247: You can use `not` or `null`
- age.el#L883: Prefer `add-to-list` to add to auto-mode-alist
- age.el#L1272: Prefer `add-to-list` to add to auto-mode-alist
- age.el#L76: Loading a package should rarely add advice
- age.el#L77: Loading a package should rarely add advice
```
The last two (on lines 76 and 77) are probably *okay* but I personally like to wait until someone's at least enabled the mode to start advising functions.  The less you shake up the runtime on a simple package import, the easier it is for users to debug their configurations.

Package and license:
- Add a LICENSE file to the repository
  See: https://github.com/licensee/licensee

anticomputer commented:
@riscy thank you for the review, I've addressed your feedback in the latest version of main and release v0.1.4 of age.el. I've also added `melpazoid` to the CI for the project, very convenient! Cheers.

riscy commented:
Looks good.  Thanks!
